common/persistence/persistence_adaptor.py

```diff
@@ -35,3 +35,12 @@ async def delete(self, key: str) -> Optional[dict]:
         :return: The instance of the item which has been deleted from persistence. (None if no item found)
         """
         pass
+
+    @abc.abstractmethod
+    async def update(self, key: str, data: dict) -> Optional[dict]:
+        """
+        Updates an item from a table given it's key and update data.
+        :param key: The key of the item to delete.
+        :return: The instance of the item which has been deleted from persistence. (None if no item found)
```

Comment:
delete?

Reply:
@mattd-kainos , could you comment on this one?

Reply:
as above, delete. this in unused

Reply:
I meant that this method is named `update` but description talks about deleting

Reply:
method deleted